### PR TITLE
Cult Update

### PR DESCRIPTION
**Pull request by Robustin**
:cl: Robustin
balance: The Cult Master's summon will no longer function in Nar-Sie Summoning Areas. 
add: Cult structures have improved audio/visual feedback and can now convert 10 runed metal into a "relic". The relics are the Bastard Sword, Xray Blindfold, and Elder Blood (worth 500 charge for Blood Rites). 
fix: All mob types should lose the blood halo upon deconversion.
balance: From 12min. to 40min. shift-time, if the cult is doing very poorly, it becomes possible for a late-arrivals to join as cultists. The late cultists will only show up once every 5 minutes at most and there is a cap (ranging from 1-4 depending on server pop.) on latejoin cultists.
balance: The shackle spell will now silence for 18 seconds, up from 10. 
balance: The unnaturally red eyes and blood halo will now appear when the cult makes up 25% and 45% of the living crew population, up from 20% and 40%. 
balance: The Blood Spear and Blood Bolts now cost 100/250 charge, down from 150/300. 
/:cl:

I designed this update to give the cult a little bit of love since they got slammed with nerfs (on top of my own) while I was gone. Their winrate since my nerfs were merged is ~30%. It also removes the kinda cheesy usage of the cult master summon for Nar-Sie. It also felt fair since I gave the cult an anti-snowball mechanic, that they get a mechanic to mitigate getting snowballed themselves. A latejoin "slot" for the cultists can only open up after 12 minutes if the cult is less than 10% of the living player population, and only one slot will open. For every 5 minutes thereafter if the <10% condition is still met, a single slot will open. The mode will track how many people joined in such a manner and will not open more slots if it would mean that latejoin cultists make up more than 5% (1-4) of the living non-cultist population. 

I decided on this approach because it strictly limits the conditions and pacing that a latejoin cultist can arrive in, ultimately making it the fairest approach for the cult to try and recover from an early loss without resorting to gimmicks. If the cult continues to be suppressed then the "latejoin" cap would be hit within 17-27 minutes into the round, negating concerns from security about having to deal with an endless stream of cultists. 

Still need to run some tests before this is merge ready though.

### CODE REVIEW

**Comment by Potato-Masher** [comment]
Very nice. What's the reason behind the Cult Master summon restriction?

**Comment by HydroFloric** [comment]
yeah the cult master summon thing seems kinda forced. It doesn't really make much sense on why it would stop it and as @Potato-Masher said it wouldn't really stop them from summoning the cult. It sounds more like a mild inconvenience.

File: code/modules/antagonists/cult/cult.dm [hunk 1]
```diff
@@ -131,7 +131,9 @@
 		H.eye_color = initial(H.eye_color)
 		H.dna.update_ui_block(DNA_EYE_COLOR_BLOCK)
 		H.remove_trait(CULT_EYES)
-		H.update_body()
+	current.cut_overlays()
+	current.regenerate_icons()
```
Comment (Razharas) [inline code/modules/antagonists/cult/cult.dm]: i cri eritiem

**Comment by Robustin** [comment]
> 
> 
> Very nice. What's the reason behind the Cult Master summon restriction? It just seems unnecessary to me when the Cult Master could just do it one tile outside the summon area (say in maint next to the area for instance).

It just bothers me when you've got a cult master dropping 20 dudes into their lap just as they're finishing the NarSie rune too. Moving the cult even a "few tiles" over can mean the difference between victory/defeat. Id rather encourage the master to use his summon in other contexts or at least before the Nar'Sie rune is attempted.

File: code/modules/antagonists/cult/cult.dm (hunk 1 above)
Comment (Robustin) [reply code/modules/antagonists/cult/cult.dm]: The blood halo appears for Xeno-Cultists and other wacky non-humans too, update_body doesn't work and even if it did it wouldn't cover all the possible cases.

**Comment by Potato-Masher** [comment]
Alright, makes sense now that I remember those instances. 

One more question. Do you consider Juggernauts to be in a good place balance wise? In my recent experience Juggernauts have been... lacking in comparison to a Wraith. I know Jugs are better suited for defense and breaking shit, but they have a hard time defending anything these days from what I have  experienced with their reduced health and resistance. 

Here's one example, I step out of the Cult base in maint to tank some shots for my brethren cultists that were taking fire, and try to scare off the Sec officer attacking. I fire my Gauntlet Echo thingy and charge forward. And then...
I die to two blasts of buckshot. Projectile I fired finally catches up and the officer just kinda sidesteps it.

I've experienced this and seen more instances similar to this. Jugs are kinda sad these days, and I much prefer making Wraiths just because their average damage + decent speed + phase shift combination is way more viable to the point that their offense makes a better defense than a Juggernaut. 

Still, this is just what I have experienced and I'd like to hear what others think about it before pressing further. Might just be my personal experience with Jugs.

**Comment by Robustin** [comment]
Since I made it harder to get shells I haven't fought enough constructs to have a good sense of where they are now. Between their barrier rune, forcewall, and the gauntlet they should be unparalleled at zone control in tight space. Getting buck-shat-on sucks but shotguns are meant to be the best counter for a cult that relies on numbers and close-range attacks (I actually added shotgun ammo to the Sec Lathe as part of an effort to make it easier to stop constructs).

If I were playing a juggernaut I would go into survival mode if I encountered buckshot, drop a forcewall, back off, find a chokepoint, drop barrier, then try to get help or escape. 

Wraiths face the same issue too, they can dodge gunfire easier but if the shotgunner is patient and waits for the wraith to attempt an attack, the wraith can die in 1-shot. I can understand your sentiment on preferring wraiths because they are more versatile - and their role as a potent saboteur cannot be overstated - but I'm OK with Juggernauts being more of a specialist. Keep in mind that even an assistant with a welding tool can beat a Wraith 1v1 (wraith dies in 4 hits), if you are trying to defend a maint base, the wraith is going to get its ass kicked in a 1-width corridor. 

What remains to be seen is whether the cult meta has truly evolved in the past several months. All these nerfs probably went too far (cult still had <50% win rate even in their "OP" phase when I first replaced talismans). My design philosophy for cult (and modes in general) is that I just try to remove the anti-fun mechanics and buff them elsewhere (with a fairer, more engaging mechanic) to compensate, but without more rounds it's hard for me to tell where the "anti-fun" exists for both sides now.

**Comment by Arianya** [comment]
>balance: Past the 12min. mark if the cult is doing sufficiently poorly, it becomes possible for a late-arrivals to join as cultists. The late cultists will only show up once every 5 minutes at most and there is a cap (ranging from 1-4 depending on server pop.) on latejoin cultists.

This is just doing the same task as mulligan but doing it poorly.

If the cult fucks up let it die, we don't need a antagonist type that can resurrect from literal total defeat for no reason.

**Comment by Robustin** [comment]
A capped trickle of latejoin cultists is nothing like "Oh you killed the Wizard? 20% of the crew is now traitors, have fun!"

If people want the round to instantly end when the cult dies off, then that would be a different story. The Cult v. Crew fight is asymmetrical in countless ways but if the cult is getting stomped while you've got a HOP opening chaplain slots... it's perfectly fair for the cult to get a couple latejoiners. 

The only other accomodation I could make it adding a time limit for the mechanic so that it won't happen after X minutes, i.e. lategame cult is wiped out after attempting to summon.

**Comment by Potato-Masher** [comment]
Alright, thanks for giving your thoughts on where Blood Cult is going. Nice to see that eliminating anti-fun is high on the list of priorities.

Nice new items, should give some endgame options for Cult. 

<s>**(AND THE RETURN OF SPIN TO WIN YEE HAW**)</s>

**Comment by Robustin** [comment]
New items are kind of an experiment, my biggest concern is that maps like Delta still have 500 plasteel lying around, otherwise I think the 10 runed metal requirement should keep the relics sufficiently limited.

File: code/game/gamemodes/cult/cult.dm [hunk 2]
```diff
@@ -96,8 +99,38 @@
 			var/datum/antagonist/cult/C = cult_mind.has_antag_datum(/datum/antagonist/cult,TRUE)
 			if(C && C.cult_team)
 				main_cult = C.cult_team
+	addtimer(CALLBACK(src, .proc/latejoin_check), 7200)
```
Comment (Cyberboss) [inline code/game/gamemodes/cult/cult.dm]: These timer shouldn't fire it the game mode ends/etc...

File: code/game/turfs/simulated/floor/reinf_floor.dm [hunk 3]
```diff
@@ -125,7 +125,8 @@
 	desc = "The air smells strangely over this sinister flooring."
 	icon_state = "plating"
 	floor_tile = null
-	var/obj/effect/clockwork/overlay/floor/bloodcult/realappearance
+	planetary_atmos = TRUE
```
Comment (Cyberboss) [inline code/game/turfs/simulated/floor/reinf_floor.dm]: Mention this in the changelog

File: code/modules/antagonists/cult/blood_magic.dm [hunk 4]
```diff
@@ -721,6 +724,13 @@
 				user.Beam(M,icon_state="sendbeam",time=10)
 		if(istype(target, /obj/effect/decal/cleanable/blood))
 			blood_draw(target, user)
+		if(istype(target, /obj/item/elder_blood))
+			user.Beam(target,icon_state="drainbeam",time=10)
+			new /obj/effect/temp_visual/cult/sparks(get_turf(target))
+			playsound(get_turf(user), 'sound/magic/enter_blood.ogg', 50)
```
Comment (Cyberboss) [inline code/modules/antagonists/cult/blood_magic.dm]:
```suggestion
			playsound(user, 'sound/magic/enter_blood.ogg', 50)
```

File: code/modules/antagonists/cult/blood_magic.dm [hunk 5]
```diff
@@ -721,6 +724,13 @@
 				user.Beam(M,icon_state="sendbeam",time=10)
 		if(istype(target, /obj/effect/decal/cleanable/blood))
 			blood_draw(target, user)
+		if(istype(target, /obj/item/elder_blood))
+			user.Beam(target,icon_state="drainbeam",time=10)
+			new /obj/effect/temp_visual/cult/sparks(get_turf(target))
```
Comment (Cyberboss) [inline code/modules/antagonists/cult/blood_magic.dm]:
```suggestion
			new /obj/effect/temp_visual/cult/sparks(target.drop_location())
```

File: code/modules/antagonists/cult/cult_structures.dm [hunk 6]
```diff
@@ -144,11 +165,53 @@
...
+		if("Veil Walker Set")
+			pickedtype += /obj/item/cult_shift
+			pickedtype += /obj/item/flashlight/flare/culttorch
+	if(src && !QDELETED(src) && anchored && pickedtype.len && Adjacent(user) && !user.incapacitated() && iscultist(user) && cooldowntime <= world.time)
+		cooldowntime = world.time + 2400
+		icon_state = "[initial(icon_state)]_off"
+		addtimer(CALLBACK(src, .proc/recharge), 2400)
+		playsound(src, 'sound/magic/teleport_app.ogg', 50, 1)
+		for(var/N in pickedtype)
+			new N(get_turf(src))
```
Comment (Cyberboss) [inline code/modules/antagonists/cult/cult_structures.dm]:
```suggestion
			new N(drop_location())
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by subject217** [comment]
cute branch name

**Comment by subject217** [comment]
10 runed metal is a pittance for a bunch of highly powerful items. latejoin cultists are completely unnecessary, it's a conversion antag. 18 second silence is unnecessary and overtuned. bloodrites don't need buffs. threshold tweaks are fine.

**Comment by ExcessiveUseOfCobblestone** [comment]
winrate should be A metric you use to gauge balance, not THE metric. We've seen already the near 50/50 spread in somegamemodes were when they were the most desired to be removed.

**Comment by subject217** [comment]
also, a suggestion: increase the threshold for ascending from 45% to 50%.

**Comment by Magikarpet** [comment]
I don't think I've ever seen nucult lose a round short of everyone willingly throwing themselves at security and lowpop

Maybe if you count summoning in time they lose more frequently but jesus christ they're too fucking fast  and they take over the station or at least overpower people buffing them to have midround spawns and bastard swords is an insane idea and seems like the most intolerable shit possible

60% of cult rounds ive seen during high and medpop seem to hit the 40% mark by 10-15 minutes if the cultists actually care about winning

**Comment by Nilons** [comment]
no bastard sword svp

**Comment by subject217** [comment]
stale?

**Comment by optimumtact** [comment]
I honestly have no idea

**Pull request closed without merging**